### PR TITLE
Fix Garden environment detection in DotcomHelperFunction

### PR DESCRIPTION
**Pull request by lysyjan**
## Description

  Fix Garden environment detection in DotcomHelperFunctions. The previous implementation relied on is_blog_garden() and get_site_meta() — WordPress.com infrastructure-level functions that are not available in all Garden environments.
   This caused isGarden() to always return false, breaking several Garden-gated features:

  - Automation emails not set to private post status
  - Classic email templates not skipped
  - WooCommerce marketing confirmation email not registered
  - Automation editor not adapting its UI

  The fix uses the IS_COMMERCE_GARDEN constant (the standard way Garden environments identify themselves), adds garden_get_wpcloud_config() as a fallback for plan detection and site metadata, and keeps existing Atomic/WooExpress
  paths unchanged.

## Code review notes

_N/A_

## QA notes

_N/A_

## Linked PRs

_N/A_

## Linked tickets

_N/A_

## After-merge notes

_N/A_

## Tasks

- [ ] I have added a changelog entry (`./do changelog:add --type=<type> --description=<description>`)
- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [x] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

## Preview

[Preview in WordPress Playground](https://account.mailpoet.com/playground/new/branch:fix-is-garden)

_The latest successful build from `fix-is-garden` will be used. If none is available, the link won't work._

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved WP.com/Garden platform detection and plan resolution, adding fallback to cloud configuration and more resilient plan parsing so plan types are detected correctly across environments.

* **Tests**
  * Added unit tests covering platform detection, garden/cloud configuration fallbacks, plan JSON parsing edge cases, and precedence between plan sources.

### CODE REVIEW

File: mailpoet/tests/unit/WPCOM/DotcomHelperFunctionsTest.php [hunk 1]
```diff
@@ -177,4 +177,77 @@ public function testGardenPartnerReturnsValueWhenMetaValueIsValid() {
...
+
+  public function testIsDotcomReturnsFalseWhenNeitherPlatform() {
+    $dotcomHelper = $this->getMockBuilder(DotcomHelperFunctions::class)
+      ->setConstructorArgs([$this->wp])
+      ->onlyMethods(['isAtomicPlatform', 'isGarden'])
+      ->getMock();
+    $dotcomHelper->method('isAtomicPlatform')->willReturn(false);
+    $dotcomHelper->method('isGarden')->willReturn(false);
+    verify($dotcomHelper->isDotcom())->false();
+  }
```
[the remark this answers is not in the document]
Comment (lysyjan) [reply mailpoet/tests/unit/WPCOM/DotcomHelperFunctionsTest.php:188]: Good catch. Solved with commit ammend and force push

**Review by pavel-mailpoet — APPROVED** [review]
